src/features/transactions/Transactions.tsx

```diff
@@ -1,13 +1,107 @@
-import React from "react";
+import { useEffect } from "react";
 import { getEtherscanURL } from "@airswap/utils";
 import { Web3Provider } from "@ethersproject/providers";
 import { useWeb3React } from "@web3-react/core";
-import { useAppSelector } from "../../app/hooks";
-import { selectTransactions } from "./transactionsSlice";
+import { useAppSelector, useAppDispatch } from "../../app/hooks";
+import {
+  selectTransactions,
+  setTransactions,
+  TransactionsState,
+} from "./transactionsSlice";
+import { revertTransaction, mineTransaction } from "./transactionActions";
+import { getTransactionsLocalStorageKey } from "../metadata/metadataApi";
 
 export function Transactions() {
   const transactions = useAppSelector(selectTransactions);
-  const { active, chainId } = useWeb3React<Web3Provider>();
+  const dispatch = useAppDispatch();
+  const { active, chainId, library, account } = useWeb3React<Web3Provider>();
+
+  useEffect(() => {
+    // Create a flag we can set to handle wallet changing between async operations
+    let walletHasChanged = false;
+
+    // get transaction state from local storage and update the transactions
+    if (chainId && account && library) {
+      const transactionsLocalStorage: TransactionsState = JSON.parse(
+        localStorage.getItem(
+          getTransactionsLocalStorageKey(chainId!, account!)
+        )!
+      ) || { all: [] };
+
+      dispatch(setTransactions(transactionsLocalStorage));
+
+      // check from all responses if one is pending... if pending, call getTransaction
+      // to see if it was a success/failure/pending. update accordingly. if pending: wait()
+      // and poll at a sensible interval.
+      transactionsLocalStorage.all.forEach(async (tx) => {
```

Comment:
It might be worth it to abstract away the callback to its own named function (not sure if in the same file or elsewhere) to improve readability. Maybe something like `processTransactions`.